src/snac/server/neo4j/Neo4JUtil.php

```diff
@@ -339,50 +339,82 @@ public function deleteConstellation(&$constellation) {
      *
      * @param \snac\data\Constellation $from The constellation object to delete/redirect from Neo4J
      * @param \snac\data\Constellation $to The constellation object that is the target of any redirects
+     * @return boolean True if successful, false otherwise
      */
     public function redirectConstellation(&$from, &$to) {
+        if ($from == null || $from->getID() == null || $to == null || $to->getID() == null) {
+            return false;
+        }
 
         if ($this->connector != null) {
             // Find all in-relations to the from constellation
             $result = $this->connector->run("MATCH p=()-[]->(b:Identity {id: {icid}}) return p;",
                 [
-                    'icid' => $from->getID()
+                    'icid' => "{$from->getID()}"
```

Comment:
Just curious, what do the brackets and quotes (`"{  }"`) add? I see them in redirectConstellation but not elsewhere?

Reply:
I had a matching issue for some, but not all neo4J queries.  Specifically, the `$from->getID()` was sending a number into graphaware, which was then not matching any constellation nodes in tests.  (because Neo4J stuck in our id numbers as strings but then doesn't do string to int matching).  So the `"{$from->getID()}"` is string interpolation, creating a string and filling it with the ID value.  I could have tried to cast it, but it feels strange to be casting in a language that's not supposed to be typed.

Reply:
I think the interpolating is fine. I just want to make sure I understand when/where Neo4J gets tripped up with types. Thanks for explaining!

Reply:
The problem is that I don't quite know what typing issues it's getting tripped up on.  The key-value pairs we put in are strings, unfortunately.  Sometimes the graphaware client seems to be okay with them; other times it isn't.  Something to test in the near future, I guess.